src/compiler/checker.ts

```diff
@@ -6254,16 +6254,16 @@ namespace ts {
                 const parameterName = node.parameterName as Identifier;
                 return {
                     kind: TypePredicateKind.Identifier,
-                    parameterName: parameterName ? parameterName.text : undefined,
+                    parameterName: parameterName ? <string>parameterName.text : undefined,

```

Comment:
Good catch; intended to leave any such changes out of this PR, so I'll revert it.